### ISSUE TITLE
About Litemono-8m

### ISSUE THREAD

**Issue by orange-ang** [noahzn/Lite-Mono#169]
Hi, firstly thanks for your nice work. I have some problems when training litemono-8m.
Here is the results.

![Image](https://github.com/user-attachments/assets/d588a36e-917b-459a-bfee-1c02c41705e9)

And the training environment(as the requirement1.12):

# packages in environment at /home/chenziang/miniconda3/envs/mono:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main  
_openmp_mutex             5.1                       1_gnu  
blas                      1.0                         mkl  
brotli-python             1.0.9            py39h6a678d5_8  
bzip2                     1.0.8                h5eee18b_6  
ca-certificates           2024.9.24            h06a4308_0  
certifi                   2024.8.30        py39h06a4308_0  
charset-normalizer        3.3.2              pyhd3eb1b0_0  
cudatoolkit               11.3.1               h2bc3f7f_2  
cycler                    0.12.1                   pypi_0    pypi
ffmpeg                    4.3                  hf484d3e_0    pytorch
fonttools                 4.54.1                   pypi_0    pypi
freetype                  2.12.1               h4a9f257_0  
geatpy                    2.7.0                    pypi_0    pypi
gmp                       6.2.1                h295c915_3  
gnutls                    3.6.15               he1e5248_0  
idna                      3.7              py39h06a4308_0  
imageio                   2.36.0                   pypi_0    pypi
intel-openmp              2023.1.0         hdb19cb5_46306  
jpeg                      9e                   h5eee18b_3  
kiwisolver                1.4.7                    pypi_0    pypi
lame                      3.100                h7b6447c_0  
lcms2                     2.12                 h3be6417_0  
ld_impl_linux-64          2.40                 h12ee557_0  
lerc                      3.0                  h295c915_0  
libdeflate                1.17                 h5eee18b_1  
libffi                    3.4.4                h6a678d5_1  
libgcc-ng                 11.2.0               h1234567_1  
libgomp                   11.2.0               h1234567_1  
libiconv                  1.16                 h5eee18b_3  
libidn2                   2.3.4                h5eee18b_0  
libpng                    1.6.39               h5eee18b_0  
libstdcxx-ng              11.2.0               h1234567_1  
libtasn1                  4.19.0               h5eee18b_0  
libtiff                   4.5.1                h6a678d5_0  
libunistring              0.9.10               h27cfd23_0  
libwebp-base              1.3.2                h5eee18b_1  
linear-warmup-cosine-annealing-warm-restarts-weight-decay 1.0                      pypi_0    pypi
lz4-c                     1.9.4                h6a678d5_1  
matplotlib                3.5.2                    pypi_0    pypi
mkl                       2023.1.0         h213fc3f_46344  
mkl-service               2.4.0            py39h5eee18b_1  
mkl_fft                   1.3.10           py39h5eee18b_0  
mkl_random                1.2.7            py39h1128e8f_0  
ncurses                   6.4                  h6a678d5_0  
nettle                    3.7.3                hbbd107a_1  
networkx                  3.2.1                    pypi_0    pypi
numpy                     1.23.3                   pypi_0    pypi
opencv-python             4.6.0.66                 pypi_0    pypi
openh264                  2.1.1                h4ff587b_0  
openjpeg                  2.5.2                he7f1fd0_0  
openssl                   3.0.15               h5eee18b_0  
packaging                 24.1                     pypi_0    pypi
pillow                    9.5.0                    pypi_0    pypi
pip                       24.2             py39h06a4308_0  
protobuf                  3.20.1                   pypi_0    pypi
pyparsing                 3.2.0                    pypi_0    pypi
pysocks                   1.7.1            py39h06a4308_0  
python                    3.9.20               he870216_1  
python-dateutil           2.9.0.post0              pypi_0    pypi
pytorch                   1.12.1          py3.9_cuda11.3_cudnn8.3.2_0    pytorch
pytorch-mutex             1.0                        cuda    pytorch
pywavelets                1.6.0                    pypi_0    pypi
readline                  8.2                  h5eee18b_0  
requests                  2.32.3           py39h06a4308_0  
scikit-image              0.19.3                   pypi_0    pypi
scipy                     1.11.1                   pypi_0    pypi
setuptools                75.1.0           py39h06a4308_0  
six                       1.16.0                   pypi_0    pypi
sqlite                    3.45.3               h5eee18b_0  
tbb                       2021.8.0             hdb19cb5_0  
tensorboardx              2.5.1                    pypi_0    pypi
thop                      0.1.1-2209072238          pypi_0    pypi
tifffile                  2024.8.30                pypi_0    pypi
timm                      0.4.12                   pypi_0    pypi
tk                        8.6.14               h39e8969_0  
torchaudio                0.12.1               py39_cu113    pytorch
torchvision               0.13.1               py39_cu113    pytorch
typing_extensions         4.11.0           py39h06a4308_0  
tzdata                    2024b                h04d1e81_0  
urllib3                   2.2.3            py39h06a4308_0  
wheel                     0.44.0           py39h06a4308_0  
xz                        5.4.6                h5eee18b_1  
zlib                      1.2.13               h5eee18b_1  
zstd                      1.5.6                hc292b87_0 

And the opts.
{
  "data_path": "/public/datasets_neo/kitti_jpg/KITTI_raw",
  "model_name": "mo0000_640_0207",
  "split": "eigen_zhou",
  "model": "lite-mono-8m",
  "weight_decay": 0.01,
  "drop_path": 0.3,
  "num_layers": 18,
  "dataset": "kitti",
  "png": false,
  "height": 192,
  "width": 640,
  "disparity_smoothness": 0.001,
  "scales": [
    0,
    1,
    2
  ],
  "min_depth": 0.1,
  "max_depth": 100.0,
  "use_stereo": false,
  "frame_ids": [
    0,
    -1,
    1
  ],
  "profile": true,
  "batch_size": 12,
  "lr": [
    0.0001,
    5e-06,
    31,
    0.0001,
    1e-05,
    31
  ],
  "num_epochs": 30,
  "scheduler_step_size": 15,
  "v1_multiscale": false,
  "avg_reprojection": false,
  "disable_automasking": false,
  "predictive_mask": false,
  "no_ssim": false,
  "mypretrain": "/home/chenziang/litemono/pretrain_weights/lite-mono-8m-pretrain.pth",
  "weights_init": "pretrained",
  "pose_model_input": "pairs",
  "pose_model_type": "separate_resnet",
  "no_cuda": false,
  "num_workers": 12,
  "load_weights_folder": null,
  "models_to_load": [
    "encoder",
    "depth",
    "pose_encoder",
    "pose"
  ],
  "log_frequency": 250,
  "save_frequency": 1,
  "disable_median_scaling": false,
  "pred_depth_scale_factor": 1,
  "ext_disp_to_eval": null,
  "eval_split": "eigen",
  "save_pred_disps": false,
  "no_eval": false,
  "eval_out_dir": null,
  "post_process": false
}

As the results are far from the paper shows, I wonder if there is anything changed when training and I didn't notice.
Hope for your advice.

**Comment by orange-ang** [comment]
And the training loss are as below.

![Image](https://github.com/user-attachments/assets/610dc168-60bf-458e-bd46-9b25d5a32ac4)

The training is on a single NVIDIA GeForce RTX 3090.  Maybe these can help you give me practical advice.

**Comment by noahzn** [comment]
Hi, can you try setting the `lr` to "lr": [
0.0001,
5e-06,
15,
0.0001,
1e-05,
15
],

**Comment by orange-ang** [comment]
Hi, sorry to bother you. But things seem to get worse.

![Image](https://github.com/user-attachments/assets/2c6f8b65-6386-49e2-bd54-9b0ff429418f)

![Image](https://github.com/user-attachments/assets/f5e896d9-9d70-4ef4-b57b-41cae44e3e35)

![Image](https://github.com/user-attachments/assets/d5694796-4317-4e0f-b596-b82b22f3a059)

As I noticed  some people mentioned replication issues similarly，while some can reproduce it easily, is there anything related to the GPUs or package version? btw I use the python version 3.9, will that influence the results?

**Comment by noahzn** [comment]
Hi, yes, according to some users' feedback using the newer versions of PyTorch could not reproduce the results. I remember that one user said this not only happened on my repo but also affected another depth estimation repo. BTW, can you evaluate all the epochs, not only the last epoch, because the best result may be achieved in an earlier epoch.

**Comment by orange-ang** [comment]
Well, that's really strange.Actually I tried to evaluate from the 16th epoch, but as you see, the results are far from the paper shows.   Anyway, thanks for your advice, I'll try to adjust optimizer parameters to see if it can help.

> Hi, yes, according to some users' feedback using the newer versions of PyTorch could not reproduce the results. I remember that one user said this not only happened on my repo but also affected another depth estimation repo. BTW, can you evaluate all the epochs, not only the last epoch, because the best result may be achieved in an earlier epoch.

**Comment by noahzn** [comment]
If you have more issues you can re-open this one or create a new one. Now I close this issue.

**Comment by Edric-star** [comment]
Sorry that I have noticed your results were actually pretty close to the author's, may I ask what the problem is?

**Comment by orange-ang** [comment]
Well, the results were not so close to the paper,  as you can see in the papers ,the results were:

![Image](https://github.com/user-attachments/assets/1af1ff13-a489-4048-be96-0cfc5340c42d)

I guess that may be something related to the python or package versions, and I haven't found the key.
> Sorry that I have noticed your results were actually pretty close to the author's, may I ask what the problem is?

**Comment by Edric-star** [comment]
Yours were 0.105 0.818 4.589 0.893 0.964 0.983, were these right? Comparing these to author's, except for the sq Rel and RMSE

**Comment by orange-ang** [comment]
Whether 0.104,etc(as showed in my first question) or 0.105,etc as you mentioned ,they are far from the paper and I'm trying to adjust 
the training options to find the better results recently. 
 And I guess you misunderstand my question, when you use lite-mono , the results may not be so strange, that may be a little different(I remembered I've tried about one year ago). But when in litemono-8m, things changed. I wonder if you compare the results with lite-mono?


> Yours were 0.105 0.818 4.589 0.893 0.964 0.983, were these right? Comparing these to author's, except for the sq Rel and RMSE

**Comment by orange-ang** [comment]
Well,  as  I don't have permissions to reopen the issue, I'm sorry to bother you in this way. I've tried several times and haven't found the key. But recently I wonder whether there were something wrong with the pretrained model since I think the training shouldn't be so sensitive to the pytorch versions. Have you tried to train litemono-8m using the pretrained model recently? 
Hope for your reply.
> If you have more issues you can re-open this one or create a new one. Now I close this issue.

**Comment by noahzn** [comment]
> Well, as I don't have permissions to reopen the issue, I'm sorry to bother you in this way. I've tried several times and haven't found the key. But recently I wonder whether there were something wrong with the pretrained model since I think the training shouldn't be so sensitive to the pytorch versions. Have you tried to train litemono-8m using the pretrained model recently? Hope for your reply.
> 
> > If you have more issues you can re-open this one or create a new one. Now I close this issue.

Hi, no, I haven't tried the training in the last year. Can you try setting the drop_path to a smaller value, like 0.2. I remember a guy also tried the newer versions of PyTorch on ManyDepth and the result was worse. So I suspect that in the newer PyTorch, some behaviours have changed and they affect the training, for example `align_corners in upsamping`.

**Comment by orange-ang** [comment]
Thanks for your advice, I'll have a try. Btw I use the _requirements1.12.1.txt_ as you provide. So it's really strange hhhh.
Anyway, thanks for your quick reply. If you have spare time, can you also try to retrain it  to see whether it's the right pretrain model.



> > Well, as I don't have permissions to reopen the issue, I'm sorry to bother you in this way. I've tried several times and haven't found the key. But recently I wonder whether there were something wrong with the pretrained model since I think the training shouldn't be so sensitive to the pytorch versions. Have you tried to train litemono-8m using the pretrained model recently? Hope for your reply.
> > > If you have more issues you can re-open this one or create a new one. Now I close this issue.
> 
> Hi, no, I haven't tried the training in the last year. Can you try setting the drop_path to a smaller value, like 0.2. I remember a guy also tried the newer versions of PyTorch on ManyDepth and the result was worse. So I suspect that in the newer PyTorch, some behaviours have changed and they affect the training, for example `align_corners in upsamping`.

**Comment by noahzn** [comment]
It should be the correct pre-trained model as I know someone used the lite-8m model and reproduced the results. Can you also try training the 8m model without using the pre-trained weights? Just to check if you can achieve the similar results.

**Comment by orange-ang** [comment]
Thanks for your quick reply.But that's the question, litemono-8m didn't show the results without pretrain model, so it's hard to judge if there is something different in different pytorch versions. Anyway, thanks for your patience, I'll keep trying.
> It should be the correct pre-trained model as I know someone used the lite-8m model and reproduced the results. Can you also try training the 8m model without using the pre-trained weights? Just to check if you can achieve the similar results.
